### PR TITLE
manifest-tool: fix static build and update 2.0.6 -> 2.1.5

### PR DESCRIPTION
**Pull request by tricktron**
## Description of changes

- Updates manifest-tool from 2.0.6 -> 2.1.5
- Fixes static build

Supersedes https://github.com/NixOS/nixpkgs/pull/225468.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tricktron** [comment]
@ofborg build pkgsStatic.manifest-tool

File: pkgs/development/tools/manifest-tool/default.nix [hunk 1]
```diff
@@ -29,14 +28,23 @@ buildGoModule rec {
...
-
-  CGO_ENABLED = if stdenv.hostPlatform.isStatic then "0" else "1";
-  GO_EXTLINK_ENABLED = if stdenv.hostPlatform.isStatic then "0" else "1";
-  ldflags = lib.optionals stdenv.hostPlatform.isStatic [ "-w" "-extldflags" "-static" ];
-  tags = lib.optionals stdenv.hostPlatform.isStatic [ "netgo" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.gitCommit=${builtins.readFile "${src}/.git-revision"}"
```
Comment (wegank) [inline pkgs/development/tools/manifest-tool/default.nix:35]: Looks like an IFD?

Comment (tricktron) [reply pkgs/development/tools/manifest-tool/default.nix:35]: @wegank I am not really familiar with the topic of Import From Derivation (IFD).

Can you help me😃? I want to read a version file which I generated during the postFetch phase. Before I read this file with a `cat` command in the preConfigurePhase.

So is this now an IFD? Are they considered bad?

Comment (tricktron) [reply pkgs/development/tools/manifest-tool/default.nix:35]: @infinisil Could me help me understand if the above is an IFD😃?

Comment (wegank) [reply pkgs/development/tools/manifest-tool/default.nix:35]: IFD is disallowed in Nixpkgs, [see here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#import-from-derivation).

As you said, just do `ldflags+="-X main.gitCommit=$(cat .git-revision)"` in `preConfigure` instead.

Comment (tricktron) [reply pkgs/development/tools/manifest-tool/default.nix]: @wegank Thanks for the info.  I removed the IFD again as you suggested.

**Review by wegank — APPROVED** [review]

**Pull request merged**